elftools/dwarf/callframe.py

```diff
@@ -148,89 +150,35 @@ def _parse_entry_at(self, offset):
             entry_structs.initial_length_field_size())
 
         # At this point self.stream is at the start of the instruction list
-        # for this entry
-        instructions = self._parse_instructions(
-            entry_structs, self.stream.tell(), end_offset)
+        # for this entry. Instructions will be parsed on demand but stream
+        # position must be adjusted to the position just after last
+        # instruction.
+        instructions_start = self.stream.tell()
+        self.stream.seek(end_offset)
 
         if is_CIE:
             self._entry_cache[offset] = CIE(
-                header=header, instructions=instructions, offset=offset,
+                header=header,
+                stream=self.stream,
+                instructions_start=instructions_start,
+                instructions_end=end_offset,
+                offset=offset,
                 augmentation_dict=aug_dict,
                 augmentation_bytes=aug_bytes,
                 structs=entry_structs)
 
         else: # FDE
             cie = self._parse_cie_for_fde(offset, header, entry_structs)
             self._entry_cache[offset] = FDE(
-                header=header, instructions=instructions, offset=offset,
+                header=header,
+                stream=self.stream,
+                instructions_start=instructions_start,
+                instructions_end=end_offset,
+                offset=offset,
                 augmentation_bytes=aug_bytes,
                 structs=entry_structs, cie=cie)
         return self._entry_cache[offset]
 
-    def _parse_instructions(self, structs, offset, end_offset):
```

Comment:
Probably, because this function has been moved to another class. For lazy instruction parsing it's is required the function to be accessible from `CFIEntry`. But `CFIEntry` object has no reference to `CallFrameInfo`. Actually, that function has not been changed.